tests/metrics/test_metric_lightning.py

```diff
@@ -0,0 +1,79 @@
+import torch
+from tests.base.boring_model import BoringModel
+from pytorch_lightning.metrics import Metric
+from pytorch_lightning import Trainer
+
+
+class SumMetric(Metric):
+    def __init__(self):
+        super().__init__()
+        self.add_state("x", torch.tensor(0.0), dist_reduce_fx="sum")
+
+    def update(self, x):
+        self.x += x
+
+    def compute(self):
+        return self.x
+
+
+def test_metric_lightning(tmpdir):
+    class TestModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.metric = SumMetric()
+            self.sum = 0.0
+
+        def training_step(self, batch, batch_idx):
```

Comment:
shouldn't self.metric() be taking in batch_idx?

Reply:
why?

Reply:
I'm just summing all the values from the batch